### DISCUSSION TITLE
When to release the connection and when not

### DISCUSSION THREAD

**Discussion by WimVriend** [digitalpetri/ethernet-ip#50] [General]
Hello Kevin,

In the documentation we find the phrase _"We recommend using connected messaging whenever possible."_.  

The logix services example shows me how to invoke the ReadTagService and WriteTagService using connected communication. But when the requested Service is complete, the example shows _pool.release(connection);_. Surely this is not the way to do it in a real application?

What would be valid reason(s) to release the connection? Failure to read or write data?

**Comment by kevinherron** [comment]
If you’re using the pool you would always release the connection after using it. It’s the same concept as a DB connection pool. You may have many service invocations pending but only a small fixed number of connections available. Each service checks it out, executes, then returns it so the next one can use it.

Using the pool is entirely optional. You can manage the connections on your own if you want.
